app/src/pages/[resourceType]/[id].tsx

```diff
@@ -29,14 +36,49 @@ export default function ResourceIDPage({ jsonData }: InferGetServerSidePropsType
     return decode('application/elm+json', jsonData);
   }, [jsonData]);
 
+  const ctx = trpc.useContext();
+
+  const draftMutation = trpc.draft.createDraft.useMutation({
+    onSuccess: data => {
+      notifications.show({
+        title: `${jsonData.resourceType} Created!`,
+        message: `${jsonData.resourceType} successfully created from ${jsonData.resourceType}/${jsonData.id}`,
+        icon: <CircleCheck />,
+        color: 'green'
+      });
+      router.push(`/authoring/${jsonData.resourceType}/${data.draftId}`);
+      ctx.draft.getDraftCounts.invalidate();
+    },
+    onError: e => {
+      notifications.show({
+        title: `${jsonData.resourceType} Creation Failed!`,
+        message: `Attempt to create ${jsonData.resourceType} from ${jsonData.resourceType}/${jsonData.id} failed with message: ${e.message}`,
+        icon: <AlertCircle />,
+        color: 'red'
+      });
+    }
+  });
+
+  const router = useRouter();
```

Comment:
Nitpick but I would move this line to be above `draftMutation` since `draftMutation` pushes to the router but is defined before the router is